Parsers/ASimNetworkSession/Parsers/vimNetworkSessionForcePointFirewall.yaml

```diff
@@ -0,0 +1,416 @@
+        endtime:datetime=datetime(null), 
+        srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+        dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+        ipaddr_has_any_prefix:dynamic=dynamic([]),
+        dstportnumber:int=int(null), 
+        hostname_has_any:dynamic=dynamic([]), 
+        dvcaction:dynamic=dynamic([]), 
+        eventresult:string='*', 
+        disabled:bool=false) {
+    let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+    let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+    let ForcePointNetwork = CommonSecurityLog
+            | where not(disabled)
+            | where (isnull(starttime) or TimeGenerated >= starttime) 
+                and (isnull(endtime) or TimeGenerated <= endtime)
+            | where DeviceVendor=="FORCEPOINT" and DeviceProduct=="Firewall"
+            | where DeviceFacility in~ ("Inspection","Packet Filtering","File Filtering") and isnotempty(DeviceEventClassID) and DeviceEventClassID != "0"    
+            | where (isnull(dstportnumber) or (DestinationPort == dstportnumber) or (ApplicationProtocol has tostring(dstportnumber)))
+            | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any), 
+                     temp_isDstMatch=has_any_ipv4_prefix(DestinationIP,dst_or_any)
+            | extend ASimMatchingIpAddr = case(
+                                array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-", // match not requested
+                                (temp_isSrcMatch and temp_isDstMatch), "Both", // has to be checked before the individual 
+                                temp_isSrcMatch, "SrcIpAddr",
+                                temp_isDstMatch, "DstIpAddr",
+                                "No match"
+                            )
+            | where ASimMatchingIpAddr != "No match"
+            ;
+    let PacketFilteringData = ForcePointNetwork
+            | where DeviceFacility == "Packet Filtering" and DeviceEventClassID !in ("70383","70393","70734","71009","71040")
+            | lookup DeviceEventClassIDLookup_Packet on DeviceEventClassID
+            | lookup MessageLookup on Message
+            | extend DvcAction = coalesce(DvcAction_MessageLookup, DvcAction_DeviceEventClassIDLookup), 
+                     EventResult = case (Message startswith "Referred connection not known", "Failure",
+                                        coalesce(EventResult_MessageLookup, EventResult_DeviceEventClassIDLookup)), 
+                     EventSeverity = case(Message startswith "Referred connection not known", "Low",
+                                        EventSeverity_DeviceEventClassIDLookup),
+                     EventOriginalResultDetails = case(Message startswith "Referred connection not known", Message,
+                                                    EventOriginalResultDetails)
+            | where ((array_length(dvcaction) == 0) or DvcAction has_any (dvcaction))
```

Comment:
Just a note: you are correct to include it in each branch for efficiency, even though it is actually the same for all branches and can theoretically be pushed to after eh union.